conans/test/unittests/client/profile_loader/profile_loader_test.py

```diff
@@ -22,336 +20,416 @@ def create_profile(folder, name, settings=None, package_settings=None, env=None,
-        self.assertEqual(profile.options.as_list(), [('zlib:aoption', '1'), ('zlib:otheroption', '12')])
-        self.assertEqual(profile.build_requires, {"*": [ConanFileReference.loads("one/1.5@lasote/stable"),
-                                                        ConanFileReference.loads("two/1.2@lasote/stable"),
-                                                        ]})
+def test_profile_dir():
+    tmp = temp_folder()
+    txt = textwrap.dedent("""
+            [env]
+            PYTHONPATH=$PROFILE_DIR/my_python_tools
+        """)
 
-    def test_profile_dir(self):
-        tmp = temp_folder()
-        txt = '''
-[env]
-PYTHONPATH=$PROFILE_DIR/my_python_tools
-'''
+    def assert_path(profile):
+        pythonpath = profile.env_values.env_dicts("")[0]["PYTHONPATH"]
+        assert pythonpath == os.path.join(tmp, "my_python_tools").replace('\\', '/')
 
-        def assert_path(profile):
-            pythonpath = profile.env_values.env_dicts("")[0]["PYTHONPATH"]
-            self.assertEqual(pythonpath, os.path.join(tmp, "my_python_tools").replace('\\', '/'))
+    abs_profile_path = os.path.join(tmp, "Myprofile.txt")
+    save(abs_profile_path, txt)
+    profile, _ = read_profile(abs_profile_path, None, None)
+    assert_path(profile)
 
-        abs_profile_path = os.path.join(tmp, "Myprofile.txt")
-        save(abs_profile_path, txt)
-        profile, _ = read_profile(abs_profile_path, None, None)
-        assert_path(profile)
+    profile, _ = read_profile("./Myprofile.txt", tmp, None)
+    assert_path(profile)
 
-        profile, _ = read_profile("./Myprofile.txt", tmp, None)
-        assert_path(profile)
+    profile, _ = read_profile("Myprofile.txt", None, tmp)
+    assert_path(profile)
 
-        profile, _ = read_profile("Myprofile.txt", None, tmp)
-        assert_path(profile)
 
-    def test_include_order(self):
-        tmp = temp_folder()
+def test_profile_include_order():
+    tmp = temp_folder()
 
-        save(os.path.join(tmp, "profile1.txt"), "MYVAR=fromProfile1")
+    save(os.path.join(tmp, "profile1.txt"), "MYVAR=fromProfile1")
 
-        profile2 = textwrap.dedent("""
+    profile2 = textwrap.dedent("""
             include(./profile1.txt)
             MYVAR=fromProfile2
             [settings]
             os=$MYVAR
-            """)
-        save(os.path.join(tmp, "profile2.txt"), profile2)
-        profile, variables = read_profile("./profile2.txt", tmp, None)
-
-        self.assertEqual(variables, {"MYVAR": "fromProfile2",
-                                     "PROFILE_DIR": tmp.replace('\\', '/')})
-        self.assertEqual(profile.settings["os"], "fromProfile2")
+        """)
+    save(os.path.join(tmp, "profile2.txt"), profile2)
+    profile, variables = read_profile("./profile2.txt", tmp, None)
+
+    assert variables == {"MYVAR": "fromProfile2",
+                         "PROFILE_DIR": tmp.replace('\\', '/')}
+    assert profile.settings["os"] == "fromProfile2"
+
+
+def test_profile_load_absolute_path():
```

Comment:
New tests start here. All other changes are related to pytest. You can check each commit separately, I didn't mix pytest with new test, each commit follows a single change.